src/Jackett.Common/Indexers/CardigannIndexer.cs

```diff
@@ -296,63 +298,81 @@ protected string applyGoTemplateText(string template, Dictionary<string, object>
                 JoinMatches = JoinMatches.NextMatch();
             }
 
-            // handle or, and functions
-            var AndOrRegex = new Regex(@"(and|or)\s+\((\..+?)\)\s+\((\..+?)\)(\s+\((\..+?)\)){0,1}");
-            var AndOrRegexMatches = AndOrRegex.Match(template);
 
-            while (AndOrRegexMatches.Success)
+            var supportedLogicFunctions = new[]
             {
+                "and",
+                "or",
+                "eq",
+                "ne"
+            };
+            var acceptsStringLiterals = new[]
+            {
+                "eq",
+                "ne"
+            };
+            var logicStr = string.Join("|", supportedLogicFunctions.Select(Regex.Escape));
+
+            // Matches a logic function above and 2 or more of (.varname) or .varname or "string literal" in any combination
+            var logicRegex = new Regex(@$"\b({logicStr})(?:\s+(\(?\.[^\)\s]+\)?|""[^""]+"")){{2,}}");
```

Comment:
You are building and compiling the RegEx in each request. You can make all of these static constants in the class.

Reply:
It used to be that instance Regex were internally cached for instance members, so this would just lead to a lookup instead of compiling it again. [.Net-Core 5 will be bringing that back](https://devblogs.microsoft.com/dotnet/regex-performance-improvements-in-net-5/), but for now, there should be a performance improvement to move this out of the function into a static constant. I will change it.